src/GregClient/Utility/FileUtilities.cs

```diff
@@ -79,21 +79,49 @@ public static byte[] GetFileHash(FileInfo fInfo)
             return hashValue;
         }
 
+        /// <summary>
+        /// Gets the full path of the user temporary folder
+        /// </summary>
+        /// <returns>String with the full path of the user temporary folder</returns>
         public static string GetTempFolder()
         {
             return Path.GetDirectoryName(Path.GetTempPath());
         }
 
+        /// <summary>
+        /// Generates a full path under the temporary folder
+        /// </summary>
+        /// <returns>String with the full path</returns>
         public static string GetTempZipPath()
         {
             var tempFolder = GetTempFolder();
-            return Path.Combine(tempFolder, "gregPkg" + DateTime.Now.Millisecond.ToString() + ".zip");
```

Comment:
Wouldn't it be cleaner to use `DateTime.Now.Ticks` and avoid looping? https://docs.microsoft.com/en-us/dotnet/api/system.datetime.ticks?view=netframework-4.7

Reply:
Hi @mmisol 

On the JIRA ticket,  there's a link to the error, very likely to be triggered by the parallel execution of the tests and the reason to do the change. So, on that context, that's the reason I've put an "exist" validation plus the loop in order to avoid the error (File Already exist). I'm trying to keep it simple and not to do a major change on the file naming so I'm not sure if the switch to ticks instead of milliseconds adds value, but that's my opinion, let me know your thoughts.

Reply:
Thanks for the context @alfredo-pozo. I think parallelism may very likely be what's causing the error.

I suggested to use `Ticks` because it's closer to the original intent of the code. It should be safe too, as according to the link I pasted `A single tick represents one hundred nanoseconds or one ten-millionth of a second.` so the chances of collision are slim to none.

Just mentioning for you to consider.

Reply:
Should this just create a guid?  There's a race condition here where multiple threads can create the same filename in the same millisecond, but not detect that they've done it because they don't create the file.

Reply:
If there's no objection, from the multiple options, I prefer the option of generate GUID's as a postfix for the names and keep the check as an additional safeguard. Any thoughts @mjkkirschner ?

Reply:
I would agree with the guid option.